apps/learner/src/routes/content/[id]/quiz/+page.svelte

```diff
@@ -61,19 +97,68 @@
         >
           <span
             class={[
-              'flex items-center justify-center rounded-lg bg-slate-100 px-2 py-1 text-xs font-semibold text-black',
+              'rounded-lg bg-slate-100 px-2 py-1 text-xs font-semibold text-black',
               selectedOptionIndex === index && '!bg-slate-900 !text-white',
             ]}
           >
             {getOptionLetter(index)}
           </span>
-          <span class="text-left text-sm">
+          <span class="text-left text-base">
             {option}
           </span>
         </button>
       {/each}
     </div>
   </div>
 
-  <Button onclick={nextQuestion} disabled={selectedOptionIndex === -1}>Check Answer</Button>
+  <Button
+    onclick={() => {
+      checkAnswer();
+      showFeedbackModal();
+    }}
+    disabled={selectedOptionIndex === -1 || isFeedbackModalOpen}>Check Answer</Button
+  >
+
+  <div
+    class="fixed inset-0 flex items-end justify-center {isFeedbackModalOpen
+      ? 'visible'
+      : 'invisible'}"
+  >
+    <div
+      class="inset-shadow-sm flex max-h-[70vh] w-full max-w-5xl transform flex-col gap-y-5 rounded-t-3xl bg-white p-5 shadow-lg transition-all {isFeedbackModalOpen
+        ? 'translate-y-0 opacity-100'
+        : 'translate-y-full opacity-0'}"
+    >
+      <div class="flex items-center justify-between">
+        <span class="py-2.5 text-xl/7 font-medium"
+          >{isCorrectAnswer ? 'Correct answer!' : 'Not quite right!'}</span
+        >
+        <button
+          class="cursor-pointer rounded-full bg-slate-100 px-2.5 py-3"
+          onclick={() => (isFeedbackModalOpen = false)}
```

Comment:
@kellylimmm 

See if we want to always have explicit functions to handle events, or if we’re okay with using inline functions.

My take is to always use explicit functions. It helps us quickly see what kind of events the page or component is handling. The downside is we’ll need to scroll up and down when reading the HTML and want to check what the function actually does, but I think the clarity is worth the extra boilerplate.